src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -197,16 +202,29 @@
  * </tbody>
  * </table></blockquote>
  * <p>
- * All the native linker implementations limit the function descriptors that they support to those that contain
- * only so-called <em>canonical</em> layouts. A canonical layout has the following characteristics:
+ * All the native linker implementations can only operate on a subset of memory layouts, called <em>supported layouts</em>.
```

Comment:
I revamped this section as I realized that what we had did not cover things in the recursive case - e.g. a struct layout is only supported if it contains other supported layouts. This new text should hopefully capture everything in a more mathematical form.